KitePrintSDK/src/main/java/ly/kite/print/Asset.java

```diff
@@ -74,6 +92,120 @@ public static MimeType fromString(String mimeType) {
+//   * Creates and returns a new asset from a bitmap. Due
+//   * to the way assets are used (i.e. they are uploaded
+//   * via HTTP, we have to encode the bitmap to (e.g.) JPEG
+//   * and store it that way.
+//   *
+//   *****************************************************/
+//  public static Asset create( Bitmap bitmap )
+//    {
+//    // Compress the bitmap into JPEG format and write into
+//    // a byte buffer.
+//
+//    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+//
+//    bitmap.compress( Bitmap.CompressFormat.JPEG, BITMAP_TO_JPEG_QUALITY, byteArrayOutputStream );
+//
+//
+//    // Create a new asset from the JPEG data
+//    return ( new Asset( byteArrayOutputStream.toByteArray(), MimeType.JPEG ) );
+//    }
+
+
+  /*****************************************************
+   *
+   * Clears any cached asset image files.
+   *
+   *****************************************************/
+  public static void clearCachedImages( Context context )
+    {
+    // Get the image cache directory
+
+    String imageCacheDirectoryPath = ImageLoader.getInstance( context ).getImageDirectoryPath( IMAGE_CLASS_STRING_ASSET );
+
+    File imageCacheDirectory = new File( imageCacheDirectoryPath );
+
+
+    // Go through all the files and delete them
+
+    File[] imageFiles = imageCacheDirectory.listFiles();
+
+    if ( imageFiles != null )
+      {
+      for ( File imageFile : imageCacheDirectory.listFiles() )
+        {
+        imageFile.delete();
+        }
+      }
+    }
+
+
+  /*****************************************************
+   *
+   * Creates a new asset from a bitmap, but writes it out
+   * to a file. The file path is automatically generated.
+   *
+   *****************************************************/
+  public static Asset createAsCachedFile( Context context, Bitmap bitmap )
+    {
+    // Generate a random file name within the cache
+    Pair<String,String> imageDirectoryAndFilePath = ImageLoader.getInstance( context ).getImageDirectoryAndFilePath( IMAGE_CLASS_STRING_ASSET, UUID.randomUUID().toString() );
+
+    File imageDirectory = new File( imageDirectoryAndFilePath.first );
+
+    imageDirectory.mkdirs();
+
+    return ( createAsCachedFile( bitmap, imageDirectoryAndFilePath.second + "." + Asset.JPEG_FILE_EXTENSION_PRIMARY ) );
+    }
+
+
+  /*****************************************************
+   *
+   * Creates a new asset from a bitmap, but writes it out
+   * to a file.
+   *
+   *****************************************************/
+  public static Asset createAsCachedFile( Bitmap bitmap, String filePath )
+    {
+    // Write the bitmap to the file
+
+    FileOutputStream fos = null;
+
+    try
+      {
+      fos = new FileOutputStream( filePath );
+
+      bitmap.compress( Bitmap.CompressFormat.JPEG, BITMAP_TO_JPEG_QUALITY, fos );
+
+      return ( new Asset( filePath ) );
+      }
+    catch ( Exception exception )
+      {
+      Log.e( LOG_TAG, "Unable to write bitmap to file" );
```

Comment:
How likely is it that writing the Bitmap will fail in practice? You probably want to add this case to your tidy up todos